numba_cuda/numba/cuda/tests/cudapy/test_inspect.py

```diff
@@ -37,28 +47,47 @@ def foo(x, y):
         # Signature in annotation
         self.assertIn("(float32, int32)", typeanno)
         file.close()
-        # Function name in LLVM
-        llvm = foo.inspect_llvm(sig)
-        self.assertIn("foo", llvm)
 
-        # Kernel in LLVM
-        self.assertIn("define void @", llvm)
-
-        asm = foo.inspect_asm(sig)
-
-        # Function name in PTX
-        self.assertIn("foo", asm)
-        # NVVM inserted comments in PTX
-        self.assertIn("Generated by NVIDIA NVVM Compiler", asm)
+        filecheck = FileCheckKernel(foo)
+        self.assertTrue(filecheck.check_llvm(sig))
+        self.assertTrue(filecheck.check_asm(sig))
 
     def test_polytyped(self):
         @cuda.jit
         def foo(x, y):
+            """
+            // LLVM: define void
+            // LLVM-SAME: foo
```

Comment:
From the doc they use `CHECK:` everywhere, so I will interpret `LLVM-x` the same as `CHECK-x`. Please correct me if I'm correct about these lines:

You first use `CHECK: define void` to find the kernel signature declaration. Then use `CHECK-SAME: foo` to check that `foo` exists in the same line as `define void`. (Regardless of whether the kernel name is mangled or not, as long as `foo` exists, this check passes).

The following lines are definitions of directives that belongs to two separate FileCheck runs. One with `LLVM_INT` and one with `LLVM_FLOAT` prefix, corresponds to the two kernel launches respectively.

Reply:
Yes, that is exactly right. The lines with only `LLVM` are shared between all tests that use the LLVM prefix, and the LLVM_INT and LLVM_FLOAT checks are specific to that datatype. In this way, we can interpolate the checks that should run for the IR with a particular signature into the same list of checks.